zrml/prediction-markets/src/tests.rs

```diff
@@ -3431,6 +3431,187 @@ fn on_resolution_correctly_reserves_and_unreserves_bonds_for_permissionless_mark
+        let end = 100;
+        assert_ok!(PredictionMarkets::create_market(
+            Origin::signed(ALICE),
+            BOB,
+            MarketPeriod::Block(0..100),
+            get_deadlines(),
+            gen_metadata(2),
+            MarketCreation::Permissionless,
+            MarketType::Categorical(2),
+            MarketDisputeMechanism::SimpleDisputes,
+            ScoringRule::CPMM,
+        ));
+        // Reserve a sentinel amount to check that we don't unreserve too much.
+        assert_ok!(Balances::reserve_named(
+            &PredictionMarkets::reserve_id(),
+            &CHARLIE,
+            SENTINEL_AMOUNT
+        ));
+        let charlie_balance_before = Balances::free_balance(&CHARLIE);
+        assert_eq!(Balances::reserved_balance(&CHARLIE), SENTINEL_AMOUNT);
+        let market = MarketCommons::market(&0).unwrap();
+        let grace_period = end + market.deadlines.grace_period;
+        let report_at = grace_period + market.deadlines.oracle_duration + 1;
+        run_to_block(report_at);
+
+        assert!(market.bonds.outsider.is_none());
+        assert_ok!(PredictionMarkets::report(
+            Origin::signed(CHARLIE),
+            0,
+            OutcomeReport::Categorical(1)
+        ));
+        let market = MarketCommons::market(&0).unwrap();
+        assert_eq!(market.bonds.outsider, Some(Bond::new(CHARLIE, OutsiderBond::get())));
+        assert_eq!(Balances::reserved_balance(&CHARLIE), SENTINEL_AMOUNT + OutsiderBond::get());
+        assert_eq!(Balances::free_balance(&CHARLIE), charlie_balance_before - OutsiderBond::get());
+        let charlie_balance_before = Balances::free_balance(&CHARLIE);
+        // on_resolution called
+        run_blocks(market.deadlines.dispute_duration);
+        assert_eq!(Balances::reserved_balance(&CHARLIE), SENTINEL_AMOUNT);
+        // Check that the outsider gets the OracleBond together with the OutsiderBond
+        assert_eq!(
+            Balances::free_balance(&CHARLIE),
+            charlie_balance_before + OracleBond::get() + OutsiderBond::get()
+        );
+    });
+}
+
+#[test]
+fn disputed_market_results_in_rewarding_outsider() {
+    ExtBuilder::default().build().execute_with(|| {
+        let end = 100;
+        let alice_balance_before = Balances::free_balance(&ALICE);
+        assert_ok!(PredictionMarkets::create_market(
+            Origin::signed(ALICE),
+            BOB,
+            MarketPeriod::Block(0..100),
+            get_deadlines(),
+            gen_metadata(2),
+            MarketCreation::Permissionless,
+            MarketType::Categorical(2),
+            MarketDisputeMechanism::SimpleDisputes,
+            ScoringRule::CPMM,
+        ));
+
+        let outsider = CHARLIE;
+        // Reserve a sentinel amount to check that we don't unreserve too much.
+        assert_ok!(Balances::reserve_named(
+            &PredictionMarkets::reserve_id(),
+            &outsider,
+            SENTINEL_AMOUNT
+        ));
```

Comment:
Random thought: Can we make it part of the mock's genesis to reserve `SENTINEL_AMOUNT` from everyone's balances?

Reply:
```
	#[pallet::genesis_build]
	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
		fn build(&self) {
			let total = self.balances.iter().fold(Zero::zero(), |acc: T::Balance, &(_, n)| acc + n);
			<TotalIssuance<T, I>>::put(total);

			<StorageVersion<T, I>>::put(Releases::V2_0_0);

			for (_, balance) in &self.balances {
				assert!(
					*balance >= <T as Config<I>>::ExistentialDeposit::get(),
					"the balance of any account should always be at least the existential deposit.",
				)
			}

			// ensure no duplicates exist.
			let endowed_accounts = self
				.balances
				.iter()
				.map(|(x, _)| x)
				.cloned()
				.collect::<std::collections::BTreeSet<_>>();

			assert!(
				endowed_accounts.len() == self.balances.len(),
				"duplicate balances in genesis."
			);

			for &(ref who, free) in self.balances.iter() {
				assert!(T::AccountStore::insert(who, AccountData { free, ..Default::default() })
					.is_ok());
			}
		}
	}
```

> T::AccountStore::insert(who, AccountData { free, ..Default::default() })

This sets `reserved` to Default::default(), which is zero. So I hoped, that we could configure that, but we can't. So we need to set it in the executable environment. This means the smallest would be a method, which reserves the sentinel amount for each test case. We could do that, but I am also okay to do it only, if we want to check against reserves in the test.

Reply:
b85409cce1a3c14c6d8398154ed7705c10c38a11

Reply:
Ah, that's too bad. I think abstracting them into a function is a good solution.

"Just one more thing..." If we correctly place `debug_assert` calls, do we even need the sentinel reserves anymore?

Reply:
Good thought. The reason for sentinel reserves is to ensure, that we recognise if we unreserve or slash too much than expected. You might mean to put `debug_assert` calls into the runtime code, not tests, otherwise you could just use normal asserts. Yes, you could place `debug_assert` inside the runtime code to check if we even have this much of funds we want to unreserve or slash.

Reply:
But you could also do it the way we already have it ;-)